src/main/java/org/spdx/utility/compare/LicenseCompareHelper.java

```diff
@@ -843,81 +844,173 @@ private static int findOriginalStart(int position, List<Pair<Integer, Integer>>
+	 * @param text    The text to search within (should not be null)
+	 * @param exception The standard SPDX license exception to search for (should not be null)
+	 * @return True if the license exception is found within the text, false otherwise (or if either argument is null)
+	 */
+	public static boolean isStandardLicenseExceptionWithinText(String text, ListedLicenseException exception) {
+		boolean result = false;
+
+		if (text == null || text.isEmpty() || exception == null) {
 			return false;
 		}
-		Pattern licenseMatchPattern = nonOptionalTextToStartPattern(nonOptionalText, CROSS_REF_NUM_WORDS_MATCH);
-		List<Pair<Integer, Integer>> charPositions = new ArrayList<>();
-		String normalizedText = normalizeText(text);
-		String compareLicenseText;
+
 		try {
-			compareLicenseText = normalizeTokensForRegex(normalizedText, charPositions);
-		} catch (IOException e1) {
-			// Just use the straight normalized license text
-			compareLicenseText = normalizeText(text);
-			charPositions.add(new ImmutablePair<>(0, 0));
+			String completeText = findTemplateWithinText(text, exception.getLicenseExceptionTemplate());
+			if (completeText != null) {
+				result = !isTextStandardException(exception, completeText).isDifferenceFound();
+			}
+		} catch (SpdxCompareException e) {
+			logger.warn("Error getting optional text for license exception ID " + exception.getLicenseExceptionId(), e);
+		} catch (InvalidSPDXAnalysisException e) {
+			logger.warn("Error getting optional text for license exception ID " + exception.getLicenseExceptionId(), e);
 		}
-		
-		Matcher matcher = licenseMatchPattern.matcher(compareLicenseText);
-		if(matcher.find()) {
-			startIndex = findOriginalStart(matcher.start(), charPositions);
-			endIndex = findOriginalStart(matcher.end(), charPositions);
-			String completeText = normalizedText.substring(startIndex, endIndex);
-			try {
-				result = !isTextStandardLicense(license, completeText).isDifferenceFound();
-			} catch (SpdxCompareException e) {
-				logger.warn("Compare exception for license ID "+license.getLicenseId(), e);
-				result = false;
-			} catch (InvalidSPDXAnalysisException e) {
-				logger.warn("Compare exception for license ID "+license.getLicenseId(), e);
-				result = false;
+
+		return result;
+	}
+
+
+	/**
+	 * Returns a list of SPDX Standard License ID's from the provided list that were found in the text, using
+	 * the SPDX matching guidelines.
+	 * @param text Text to compare to the standard license texts
+	 * @param licenseIds License ids to compare against
+	 * @return List of SPDX standard license IDs from licenseIds that match
+	 * @throws InvalidSPDXAnalysisException If an error occurs accessing the standard licenses
+	 * @throws SpdxCompareException If an error occurs in the comparison
+	 */
+	public static List<String> matchingStandardLicenseIds(String text, List<String> licenseIds) throws InvalidSPDXAnalysisException, SpdxCompareException {
```

Comment:
Indeed - this function will be very slow due to the number of regular expressions being process.

One way to speed this up is to narrow the list of possible matching license ID's using a more efficient technique.  The LicenseMatcher code uses this approach with a Sorenson Dice algorithm - see the [matcher code](https://github.com/spdx/spdx-license-matcher/blob/master/spdx_license_matcher/matcher.py) for an example.